ai_ref_kits/multimodal_ai_visual_generator/ci/test.py

```diff
@@ -38,22 +39,29 @@
     "MODEL_PRECISION": PRECISION
 })
 
-process = subprocess.Popen(
-    [sys.executable, "-m", "uvicorn", "main:app", "--host", "127.0.0.1", "--port", "8000"],
-    env=env
-)
+main_path = Path(__file__).resolve().parent.parent / "main.py"
+
+process = subprocess.Popen([
+    sys.executable,
+    "-m", "uvicorn",
+    f"{main_path.stem}:app",
+    "--app-dir", str(main_path.parent),
+    "--host", "127.0.0.1",
+    "--port", "8000"
+], env=env)
 
 try:
-    # Wait up to ~130 seconds (130 retries x 1s sleep) for FastAPI server to come up
-    for _ in range(130):
+    # Wait for FastAPI to become responsive
+    retries = 1000 if platform.system() == "Darwin" else 130

```

Comment:
capture subprocess output and wait for e.g. running on localhost. instead of active requesting

Reply:
Updated the test to capture stdout in gradio_log.txt and wait for the "Uvicorn running on" message in logs, replacing the earlier polling approach.